### PR TITLE
OCR_PFE_ASSIGNEMENT

### PR DESCRIPTION
**Pull request by Zac-here**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1 +1,57 @@
...
+
+
+def getImageFromPath(image_path):
+    # Read image
+    image = keras_ocr.tools.read(image_path)
+
+    return image
+
+
+def getWordsList(prediction_groups):
```
Comment (omarmhaimdat) [inline src/main.py:16]: This whole function could be replaced by one simple line:
```python
return [pred[0] for pred in prediction_groups[0]]
```

**Review by omarmhaimdat — COMMENTED** [review]
The code is small and simple, but I do have some comments on it:

- Naive usage of external libraries. I think some effort could have been put to fit the specificities of the input data.
- No multi-language support. The code is only able to recognize english words.
- No error handling. The code is not able to handle errors and exceptions.
- No effort to post-process the results.
- Non conventional code style. The code is following camelCase convention, which is not the standard in python.
- No efforts to validate the results. The code is not able to validate the results, and is not able to handle the case where the OCR fails to recognize the words.
  - CER and WER are not calculated as requested in the intructions.

# Conclusion

In conclusion, I think there is a lot of room for improvement in this submission. I think the code could have been improved by:
- Fitting the OCR to the specificities of the input data.
- Adding multi-language support.
- Adding error handling.
- Adding post-processing.
- Adding validation.
- Adding a better code style.

**Comment by omarmhaimdat** [comment]
I will close this pull request, you will receive an answer by email on Monday.
